### PR TITLE
Grouping tests

### PR DESCRIPTION
**Pull request by amanmahajan7**

### CODE REVIEW

File: test/grouping.test.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      selectedRows={selectedRows}
+      onSelectedRowsChange={setSelectedRows}
+      expandedGroupIds={expandedGroupIds}
+      onExpandedGroupIdsChange={setExpandedGroupIds}
+    />
+  );
+}
+
+function setup(groupBy?: string[]) {
+  return render(
```
Comment (nstepien) [inline test/grouping.test.tsx:80]: Doesn't really matter, you can ignore this if you want
```suggestion
  render(
```

File: test/grouping.test.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+test('should group by single column', () => {
+  setup(['country']);
+  expect(queryGrid()).not.toBeInTheDocument();
+  expect(getTreeGrid()).toHaveAttribute('aria-rowcount', '7');
+  expect(getRows()).toHaveLength(2);
+});
+
+test('should group by multiple columns', () => {
+  setup(['country', 'year']);
```
Comment (nstepien) [inline test/grouping.test.tsx:108]: Should we add a test where we group the same column multiple times?

File: test/grouping.test.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  expect(getTreeGrid()).toHaveAttribute('aria-rowcount', '11');
+  expect(getRows()).toHaveLength(2);
+});
+
+test('should use column order while grouping', () => {
```
Comment (nstepien) [inline test/grouping.test.tsx:113]:
```suggestion
test('should use groupBy order while grouping', () => {
```

File: test/grouping.test.tsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+  setup(['year', 'country']);
+  expect(getTreeGrid()).toHaveAttribute('aria-rowcount', '12');
+  expect(getRows()).toHaveLength(3);
```
Comment (nstepien) [inline test/grouping.test.tsx:116]: Could test the header cell are in the correct order

File: test/grouping.test.tsx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+});
+
+test('should toggle group when group cell is clicked', () => {
+  setup(['year']);
+  expect(getRows()).toHaveLength(3);
+  expect(getRows()).toHaveLength(3);
```
Comment (nstepien) [inline test/grouping.test.tsx:122]:
```suggestion
  expect(getRows()).toHaveLength(3);
```

File: test/grouping.test.tsx [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+  userEvent.click(within(groupCell1.parentElement!).getByLabelText('Select Group'));
+  let selectedRows = screen.getAllByRole('row', { selected: true });
+  expect(selectedRows).toHaveLength(4);
+  expect(selectedRows[0]).toHaveAttribute('aria-rowindex', '5');
+  expect(selectedRows[1]).toHaveAttribute('aria-rowindex', '6');
+  expect(selectedRows[2]).toHaveAttribute('aria-rowindex', '8');
+  expect(selectedRows[3]).toHaveAttribute('aria-rowindex', '9');
+
+  // unselecting child should unselect the parant row
+  //   userEvent.click(selectedRows[3].children[0]);
```
Comment (nstepien) [inline test/grouping.test.tsx:186]:
```suggestion
```

**Review by nstepien — CHANGES_REQUESTED** [review]

File: test/grouping.test.tsx [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+  // select parent row
+  userEvent.click(within(groupCell1.parentElement!).getByLabelText('Select Group'));
+  let selectedRows = screen.getAllByRole('row', { selected: true });
+  expect(selectedRows).toHaveLength(4);
+  expect(selectedRows[0]).toHaveAttribute('aria-rowindex', '5');
+  expect(selectedRows[1]).toHaveAttribute('aria-rowindex', '6');
+  expect(selectedRows[2]).toHaveAttribute('aria-rowindex', '8');
+  expect(selectedRows[3]).toHaveAttribute('aria-rowindex', '9');
+
+  // unselecting child should unselect the parant row
```
Comment (nstepien) [inline test/grouping.test.tsx:185]:
```suggestion
  // unselecting child should unselect the parent row
```

**Review by nstepien — CHANGES_REQUESTED** [review]

File: test/grouping.test.tsx (hunk 2 above)
Comment (amanmahajan7) [reply test/grouping.test.tsx:114]: Done

File: test/grouping.test.tsx (hunk 4 above)
Comment (amanmahajan7) [reply test/grouping.test.tsx:130]: Done

**Comment by amanmahajan7** [comment]
I will add keyboard navigation test in the next PR

**Review by nstepien — APPROVED** [review]

**Pull request merged**